tests/utils/validate_cloud_optimized_geotiff.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# *****************************************************************************
+#  $Id$
+#
+#  Project:  GDAL
+#  Purpose:  Validate Cloud Optimized GeoTIFF file structure
+#  Author:   Even Rouault, <even dot rouault at spatialys dot com>
+#
+# *****************************************************************************
+#  Copyright (c) 2017, Even Rouault
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+#  OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+# *****************************************************************************
+
+import os.path
```

Comment:
Is there not a way to bring this in as a dependency rather than copy the code in?

Reply:
I didn't find a pip package for it, the Proper Ecosystem Thing™ would probably be to PR https://github.com/rouault/cog_validator w/ pip support. But for now this was the best way I could think of.